log-group-events-lambda/handler/secret_ops.go

```diff
@@ -0,0 +1,172 @@
+package handler
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
+	"github.com/aws/aws-secretsmanager-caching-go/secretcache"
+	"github.com/logzio/firehose-logs/common"
+	"os"
+	"regexp"
+	"sort"
+)
+
+// SecretsManagerAPIInterface AWS SDK v2 doesn't provide an interface for each service client like v1
+type SecretsManagerAPIInterface interface {
```

Comment:
Lets talk about what this interface and the structs are required

Reply:
Needed for the mocks in the test